### ISSUE TITLE
Error when running fitNULLGLMM on example dataset using Saige 0.42

### ISSUE THREAD

**Issue by aokulabasile** [weizhouUMICH/SAIGE#271]
Hello,

I am getting the following error when running step1_fitNULLGLMM.R on the example dataset that has been provided using saige 0.42:

```
Error in if (max(abs(alpha - alpha0)/(abs(alpha) + abs(alpha0) + tol.coef)) <  :
  missing value where TRUE/FALSE needed
Calls: fitNULLGLMM ... system.time -> glmmkin.ai_PCG_Rcpp_Quantitative -> Get_Coef
Timing stopped at: 0.118 0.001 0.118
```
Here is the my command and the full log:
```
Rscript /gpfs/commons/home/abasile/software/saige/SAIGE/extdata/step1_fitNULLGLMM.R --plinkFile=~/software/saige/SAIGE/extdata/in
put/nfam_100_nindep_0_step1_includeMoreRareVariants_poly --phenoFile=~/software/saige/SAIGE/extdata/input/pheno_1000samples.txt_wit
hdosages_withBothTraitTypes.txt --phenoCol=y_quantitative --covarColList=x1,x2 --sampleIDColinphenoFile=IID --traitType=quantitative --outputPrefix=example_quant --nThreads=4 --LOCO=FALSE --IsOverwriteVarianceRatioFile=TRUE
Loading required package: optparse
R version 3.6.1 (2019-07-05)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)
  
Matrix products: default
BLAS/LAPACK: /nfs/sw/anaconda3/anaconda3-10.19/envs/RSAIGE/lib/R/lib/libRblas.so
  
locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C
  
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base
  
other attached packages:
[1] optparse_1.6.4 SAIGE_0.42.1

loaded via a namespace (and not attached):
[1] compiler_3.6.1     Matrix_1.2-17      Rcpp_1.0.1         getopt_1.20.3
[5] grid_3.6.1         RcppParallel_4.4.4 lattice_0.20-38
$plinkFile
[1] "~/software/saige/SAIGE/extdata/input/nfam_100_nindep_0_step1_includeMoreRareVariants_poly"
  
$phenoFile
[1] "~/software/saige/SAIGE/extdata/input/pheno_1000samples.txt_withdosages_withBothTraitTypes.txt"
  
$phenoCol
[1] "y_quantitative"

$traitType
[1] "quantitative"
  
$invNormalize
[1] FALSE
  
$covarColList
[1] "x1,x2"
  
$sampleIDColinphenoFile
[1] "IID"
  
$tol
[1] 0.02
  
$maxiter
[1] 20
  
$tolPCG
[1] 1e-05
  
$maxiterPCG
[1] 500
  
$nThreads
[1] 4
  
$SPAcutoff
[1] 2

$numRandomMarkerforVarianceRatio
[1] 30
  
$skipModelFitting
[1] FALSE
  
$memoryChunk
[1] 2
  
$tauInit
[1] "0,0"
  
$LOCO
[1] FALSE
  
$traceCVcutoff
[1] 0.0025
  
$ratioCVcutoff
[1] 0.001
  
$outputPrefix
[1] "example_quant"
  
$IsOverwriteVarianceRatioFile
[1] TRUE
  
$IsSparseKin
[1] FALSE
  
$numRandomMarkerforSparseKin
[1] 2000

$isCateVarianceRatio
[1] FALSE
  
$relatednessCutoff
[1] 0.125
  
$cateVarRatioMinMACVecExclude
[1] "0.5,1.5,2.5,3.5,4.5,5.5,10.5,20.5"
  
$cateVarRatioMaxMACVecInclude
[1] "1.5,2.5,3.5,4.5,5.5,10.5,20.5"
  
$isCovariateTransform
[1] TRUE
  
$isDiagofKinSetAsOne
[1] FALSE
  
$useSparseSigmaConditionerforPCG
[1] FALSE
  
$useSparseSigmaforInitTau
[1] FALSE
  
$minMAFforGRM
[1] 0.01
  
$minCovariateCount
[1] -1
  
$includeNonautoMarkersforVarRatio
[1] FALSE

$help
[1] FALSE
  
tauInit  is  0 0
cateVarRatioMinMACVecExclude  is  0.5 1.5 2.5 3.5 4.5 5.5 10.5 20.5
cateVarRatioMaxMACVecInclude  is  1.5 2.5 3.5 4.5 5.5 10.5 20.5
Markers in the Plink file with MAF >=  0.01  will be used to construct GRM
4  threads are set to be used
The variance ratio file  example_quant.varianceRatio.txt  already exists. IsOverwriteVarianceRatioFile=TRUE so the file will be overwritten
1000  samples have genotypes
formula is  y_quantitative~x1+x2
1000  samples have non-missing phenotypes
1000  samples will be used for analysis
qr transformation has been performed on covariates
colnames(data.new) is  Y minus1 x1 x2
out.transform$Param.transform$qrr:  3 3
y_quantitative  is a quantitative trait
glm:

Call:  glm(formula = formula.new, family = gaussian(link = "identity"),
    data = data.new)
  
Coefficients:
    minus1          x1          x2
-9.395e-17  -5.059e-01  -2.461e-01
  
Degrees of Freedom: 1000 Total (i.e. Null);  997 Residual
Null Deviance:      998.7
Residual Deviance: 682.2        AIC: 2463
Start fitting the NULL GLMM
   user  system elapsed
  1.288   0.386   1.844
[1] "Start reading genotype plink file here"
Error! fam file not open!time: 0
[1] "Genotype reading is done"
Fixed-effect coefficients:
       minus1            x1            x2
-9.394736e-17 -5.059180e-01 -2.460621e-01
initial tau is  1 0
iGet_Coef:  1
iter from getPCG1ofSigmaAndVector 1
iter from getPCG1ofSigmaAndVector 1
iter from getPCG1ofSigmaAndVector 1
iter from getPCG1ofSigmaAndVector 1
  
warning: inv_sympd(): given matrix is not symmetric
Tau:
[1] 1 0
Fixed-effect coefficients:
     [,1]
[1,]  NaN
[2,]  NaN
[3,]  NaN
Error in if (max(abs(alpha - alpha0)/(abs(alpha) + abs(alpha0) + tol.coef)) <  :
  missing value where TRUE/FALSE needed
Calls: fitNULLGLMM ... system.time -> glmmkin.ai_PCG_Rcpp_Quantitative -> Get_Coef
Timing stopped at: 0.118 0.001 0.118
Execution halted
```
Thank you in advance for your help

**Comment by mikacashman** [comment]
Hi there,

I am also getting a similar issue which I can see has been posted a few times.  Seems related to #136 but in this case myself and the poster above me are using V0.42.1.  

Error message:
```
Error in if (abs(q - m1)/sqrt(var1) < Cutoff) { :
  missing value where TRUE/FALSE needed
Calls: fitNULLGLMM ... scoreTest_SPAGMMAT_forVarianceRatio_binaryTrait -> <Anonymous>
In addition: Warning message:
In sqrt(var1) : NaNs produced
Execution halted
```

I would also like to comment that I determined this error to be flaky for me.  Upon observing the error when I previously remembered the script to work I ran a series of them back-to-back.  It was about a 50% chance of running into this fatal error and 50% chance of success (including the resulting model that worked in Step2).

Are there any ideas or advice regarding this error on newer SAGIE versions?
Thank you

<details><summary>Click here to see further log output</summary><pre>
Loading required package: optparse
[...]
R version 3.6.1 (2019-07-05)
Platform: powerpc64le-unknown-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server 7.6 (Maipo)

Matrix products: default
BLAS/LAPACK: /autofs/nccs-svm1_sw/summit/r/3.6.1/rhel7.6_gnu6.4.0/lib64/R/lib/libRblas.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] SAIGE_0.42.1   optparse_1.6.6

loaded via a namespace (and not attached):
[1] compiler_3.6.1     Matrix_1.2-17      Rcpp_1.0.5         getopt_1.20.3
[5] grid_3.6.1         RcppParallel_5.0.2 lattice_0.20-38
$plinkFile
[...]
$traitType
[1] "binary"

$invNormalize
[1] FALSE

$covarColList
[1] "PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10,PC11,PC12,PC13,PC14,PC15,PC16,PC17,PC18,PC19,PC20"

$sampleIDColinphenoFile
[1] "IID"

$tol
[1] 0.02

$maxiter
[1] 20

$tolPCG
[1] 1e-05

$maxiterPCG
[1] 500

$nThreads
[1] 1

$SPAcutoff
[1] 2

$numRandomMarkerforVarianceRatio
[1] 30

$skipModelFitting
[1] FALSE

$memoryChunk
[1] 2

$tauInit
[1] "0,0"

$LOCO
[1] FALSE

$traceCVcutoff
[1] 0.0025

$ratioCVcutoff
[1] 0.001

$outputPrefix
[...]

$IsOverwriteVarianceRatioFile
[1] FALSE

$IsSparseKin
[1] FALSE

$numRandomMarkerforSparseKin
[1] 2000

$isCateVarianceRatio
[1] FALSE

$relatednessCutoff
[1] 0.125

$cateVarRatioMinMACVecExclude
[1] "0.5,1.5,2.5,3.5,4.5,5.5,10.5,20.5"

$cateVarRatioMaxMACVecInclude
[1] "1.5,2.5,3.5,4.5,5.5,10.5,20.5"

$isCovariateTransform
[1] TRUE

$isDiagofKinSetAsOne
[1] FALSE

$useSparseSigmaConditionerforPCG
[1] FALSE

$useSparseSigmaforInitTau
[1] FALSE

$minMAFforGRM
[1] 0.01

$minCovariateCount
[1] -1

$includeNonautoMarkersforVarRatio
[1] FALSE

$FemaleOnly
[1] FALSE

$MaleOnly
[1] FALSE

$sexCol
[1] ""

$FemaleCode
[1] "1"

$MaleCode
[1] "0"

$noEstFixedEff
[1] FALSE

$help
[1] FALSE

tauInit  is  0 0
cateVarRatioMinMACVecExclude  is  0.5 1.5 2.5 3.5 4.5 5.5 10.5 20.5
cateVarRatioMaxMACVecInclude  is  1.5 2.5 3.5 4.5 5.5 10.5 20.5
Markers in the Plink file with MAF >=  0.01  will be used to construct GRM
869  samples have genotypes
formula is  y~PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10+PC11+PC12+PC13+PC14+PC15+PC16+PC17+PC18+PC19+PC20
869  samples have non-missing phenotypes
869  samples will be used for analysis
qr transformation has been performed on covariates
colnames(data.new) is  Y minus1 PC1 PC2 PC3 PC4 PC5 PC6 PC7 PC8 PC9 PC10 PC11 PC12 PC13 PC14 PC15 PC16 PC17 PC18 PC19 PC20
out.transform$Param.transform$qrr:  21 21
y  is a binary trait
glm:

Call:  glm(formula = formula.new, family = binomial, data = data.new)

Coefficients:
  minus1       PC1       PC2       PC3       PC4       PC5       PC6       PC7
-3.86100  -0.27625  -0.57741   0.80921   0.06293  -0.10731  -0.55593  -1.71094
     PC8       PC9      PC10      PC11      PC12      PC13      PC14      PC15
-0.50464   0.25357   0.66140   0.05131  -0.38465  -0.02100  -0.66162  -0.01988
    PC16      PC17      PC18      PC19      PC20
 0.35405  -0.09095   0.32580  -0.23774  -0.02720

Degrees of Freedom: 869 Total (i.e. Null);  848 Residual
Null Deviance:      1205
Residual Deviance: 285.6        AIC: 327.6
Start fitting the NULL GLMM
   user  system elapsed
  5.865   0.045   2.892
[1] "Start reading genotype plink file here"
nbyte: 218
nbyte: 218
reserve: 172935184

M: 786069, N: 869
size of genoVecofPointers: 1
setgeno mark1
setgeno mark2
786069 markers with MAF >= 0.01 are used for GRM.
setgeno mark5
setgeno mark6
time: 16440
[1] "Genotype reading is done"
inital tau is  1 0.1
iGet_Coef:  1
iter from getPCG1ofSigmaAndVector 3
iter from getPCG1ofSigmaAndVector 3

[.........Long log cut.........]

254995 th marker
G0 1 0 0 0 0 0 1 0 0 0
iter from getPCG1ofSigmaAndVector 4
684810 th marker
G0 0 0 0 0 0 0 1 0 0 0
iter from getPCG1ofSigmaAndVector 3
[1] "OK"
[1] "OK1"
[1] "OK2"
CV for variance ratio estimate using  120  markers is  0.08629677  >  0.001
try  130  markers
35897 th marker
G0 1 0 1 1 0 1 0 1 1 1
iter from getPCG1ofSigmaAndVector 4
Error in if (abs(q - m1)/sqrt(var1) < Cutoff) { :
  missing value where TRUE/FALSE needed
Calls: fitNULLGLMM ... scoreTest_SPAGMMAT_forVarianceRatio_binaryTrait -> <Anonymous>
In addition: Warning message:
In sqrt(var1) : NaNs produced
Execution halted
</pre></details>

**Comment by weizhouUMICH** [comment]
Sorry fo the late reply! We have just released a new version 1.0.0. It has computational efficiency improvements for both Step 1 and Step 2 for single-variant and set-based tests. We have created a new program github page https://github.com/saigegit/SAIGE with the documentation provided https://saigegit.github.io/SAIGE-doc/
The program will be maintained by multiple SAIGE developers there. The docker image has been updated. Please feel free to try the version 1.0.0 and report issues if any.

Thanks!
Wei
